### PR TITLE
ci: Add notebooks to CI tests plus additional cleanup

### LINKED ISSUE #92: Add notebooks tests in CI
https://github.com/marketplace/actions/nbmake

### PR DESCRIPTION
**Pull request by mortendaehli**
# Pull Request

### WHAT

Add notebooks to CI tests

### WHY

To ensure our example-/test- notebooks are running as expected.

### HOW

Add nbmake to dev-tools and run pytest with nbmake. This will convert notebooks to pure Python and run the tests on notebooks prefixed test_. Will have to do additional updates then we can run on all notebooks. Example cases are failing. I did fix much of the failing code, but need to do the remaining in another PR.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix
- [x] DevOps, CI/CD

## What to test/verify

Pipelines should run without errors.
 
## Checklist:

- [x] :tada: This PR closes #92 .
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Task 1: Add notebooks to CI tests
   - [x] Task 2: Fix failing notebooks + chore tasks.
- [x] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGES.md`.
- [x] :books: I have considered updating the documentation in `README.md`, `userguide.md` or `docs/oogeso_manual.tex`.

### CODE REVIEW

File: tests/test_data/leogo_reference_platform.yaml [hunk 1]
```diff
@@ -101,7 +101,7 @@ devices:
   - id: battery
     E_max: 4
     flow_max: 4
-    eta: 0.9
+    eta: 0.9oogeso
```
Comment (HaraldGSvendsen) [inline tests/test_data/leogo_reference_platform.yaml:104]: feil, vel?

**Review by HaraldGSvendsen — APPROVED** [review]
Burde fjerne denne: examples/results/timerseries_ts0_Base.pdf

File: tests/test_data/leogo_reference_platform.yaml (hunk 1 above)
Comment (mortendaehli) [reply tests/test_data/leogo_reference_platform.yaml:104]: Takk takk :) Feil ja...

**Review by mortendaehli — COMMENTED** [review]
Need QA on the notebooks

**Pull request merged**